openhands/runtime/utils/runtime_build.py

```diff
@@ -197,6 +200,60 @@ def build_runtime_image(
     docker_build_folder: str | None = None,
     dry_run: bool = False,
     force_rebuild: bool = False,
+) -> str:
+    runtime_image_repo, runtime_image_tag = get_runtime_image_repo_and_tag(base_image)
+
+    generic_runtime_image_name = f'{runtime_image_repo}:{runtime_image_tag}'
+
+    # Prepare build folder only once
+    cur_docker_build_folder = docker_build_folder or tempfile.mkdtemp()
+
+    if not force_rebuild and runtime_builder.image_exists(generic_runtime_image_name):
+        from_scratch_hash = prep_docker_build_folder(
+            cur_docker_build_folder,
+            base_image=generic_runtime_image_name,
+            skip_init=True,
+            extra_deps=extra_deps,
+        )
+    else:
+        from_scratch_hash = prep_docker_build_folder(
+            cur_docker_build_folder,
+            base_image=base_image,
+            skip_init=False,
+            extra_deps=extra_deps,
+        )
+
+    hash_runtime_image_name = f'{runtime_image_repo}:{from_scratch_hash}'
+
+    if not force_rebuild and runtime_builder.image_exists(hash_runtime_image_name):
+        if docker_build_folder is None:
+            shutil.rmtree(cur_docker_build_folder)
+        return hash_runtime_image_name
+
+    if not dry_run:
+        _build_sandbox_image(
+            docker_folder=cur_docker_build_folder,
+            runtime_builder=runtime_builder,
+            target_image_repo=runtime_image_repo,
+            target_image_hash_tag=from_scratch_hash,
+            target_image_tag=runtime_image_tag,
+        )
+    else:
+        logger.info(f'Dry run: Skipping image build for [{generic_runtime_image_name}]')
+
+    if docker_build_folder is None:
+        shutil.rmtree(cur_docker_build_folder)
+
+    return hash_runtime_image_name
+
+
+def build_runtime_image_old(
+    base_image: str,
+    runtime_builder: RuntimeBuilder,
+    extra_deps: str | None = None,
+    docker_build_folder: str | None = None,
+    dry_run: bool = False,
+    force_rebuild: bool = False,
```

Comment:
I'd say if your simpler version works - we should directly get rid of the old code to prevent stale code in our code base? We can always find the older version via git anyway

Reply:
Yes, I only kept that for the comments